File: Electronics_Shop_17.Model/Requests/AddOrder.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Electronics_Shop_17.Model.Requests
{
    public class AddOrder
    {
        public DateTime OrderTime { get; set; } = DateTime.UtcNow;
        public double TotalAmount { get; set; }
        public string StateMachine { get; set; } = "Pending";

        public int CustomerId { get; set; }
        public int AdressId { get; set; }
        public int? CouponId { get; set; }
        public int PaymentMethodId { get; set; }
        public virtual List<AddOrderItem> OrderItems { get; set; }
        public string PaymentId { get; set; } = "";
        public string PaymentIntent { get; set; } = "";
    }
}

File: angular17_web/src/app/models/order/add.order.ts
import { AddOrderItem } from '../order-item/add.order.item';

export class AddOrder {
  orderTime?: Date;
  totalAmount!: number;
  stateMachine?: string;
  customerId!: number;
  adressId!: number;
  couponId?: number;
  paymentMethodId!: number;
  orderItems!: AddOrderItem[];
  paymentId?: string;
  paymentIntent?: string;

  constructor(data?: any) {
    if (!data) return;

    this.orderTime = data.orderTime ? new Date(data.orderTime) : undefined;
    this.totalAmount = data.totalAmount;
    this.stateMachine = data.stateMachine;
    this.customerId = data.customerId;
    this.adressId = data.adressId;
    this.couponId = data.couponId;
    this.paymentMethodId = data.paymentMethodId;
    this.orderItems = Array.isArray(data.orderItems)
      ? data.orderItems.map((x: any) => new AddOrderItem(x))
      : [];
    this.paymentId = data.paymentId;
    this.paymentIntent = data.paymentIntent;
  }
}
